client/src/router/Router.test.jsx

```diff
@@ -0,0 +1,100 @@
+import { act, render, screen } from "@testing-library/react";
+import { describe, it, expect, beforeEach } from "vitest";
+import { Link, Router } from "./Router";
+import { navigate } from "./navigate";
+
+describe("Router", () => {
+  beforeEach(() => {
+    // Reset the URL before each test
+    window.history.pushState({}, "", "/");
+  });
+
+  const MockHome = () => <h1>Home Page</h1>;
+  const MockSomepage = () => <h1>Somepage Page</h1>;
+  const MockLinks = () => (
+    <>
+      <h1>Links Page</h1>
+      <Link href="/">home</Link>
+    </>
+  );
+  const MockParams2 = ({ params }) => (
+    <>
+      <h1>Params</h1>
+      <div>
+        {params.first}+{params.second}
+      </div>
+    </>
+  );
+  const MockParams1 = ({ params }) => (
+    <>
+      <h1>Params</h1>
+      <div>{params.first}</div>
+    </>
+  );
+
+  const routes = [
+    { "/": MockHome },
+    { "/somepage": MockSomepage },
+    { "/links": MockLinks },
+    { "/params/:first/:second": MockParams2 },
+    { "/one/two/three/:first": MockParams1 },
```

Comment:
```suggestion
    { "/onePathElement/:first/:second": MockParams2 },
    { "/multiple/path/elements/:first": MockParams1 },
```
I got a bit confused by the namings and what was tested, do you think this is better?

Reply:
ah yeah I see what you mean, I'll make the change.